internal/config/types_defaults_test.go

```diff
@@ -0,0 +1,269 @@
+package config
+
+// Copyright (C) 2024 by Posit Software, PBC.
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/posit-dev/publisher/internal/interpreters"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/types"
+	"github.com/posit-dev/publisher/internal/util"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+)
+
+type ConfigFillDefaultsSuite struct {
+	suite.Suite
+	cwd                      util.AbsolutePath
+	log                      logging.Logger
+	rInterpreter             *interpreters.RInterpreter
+	rMissingInterpreter      *interpreters.RInterpreter
+	pythonInterpreter        *interpreters.PythonInterpreter
+	pythonMissingInterpreter *interpreters.PythonInterpreter
```

Comment:
Ditto here about interfaces and pointers, generally this isn't needed and complicates things

Reply:
The code and tests have been reworked. Thanks.